src/ol/layer/layergroup.js

```diff
@@ -123,9 +133,15 @@ ol.layer.Group.prototype.handleLayersChanged_ = function(event) {
  */
 ol.layer.Group.prototype.handleLayersAdd_ = function(collectionEvent) {
   var layer = /** @type {ol.layer.Base} */ (collectionEvent.element);
-  this.listenerKeys_[goog.getUid(layer).toString()] = goog.events.listen(
-      layer, [ol.ObjectEventType.PROPERTYCHANGE, goog.events.EventType.CHANGE],
-      this.handleLayerChange_, false, this);
+  var key = goog.getUid(layer).toString();
+  goog.asserts.assert(!goog.isDef(this.listenerKeys_[key]),
```

Comment:
Using `goog.asserts.assert(!(key in this.listenerKeys_))` would make more sense to me.